src/network/receiver.go

```diff
@@ -0,0 +1,57 @@
+package network
+
+import (
+	"net"
+	"strconv"
+	"time"
+)
+
+/*
+Listen for incoming messages on specified IP and port.
+*/
+func ListenForMessages(ip string, port int, messageChannelchan chan<- []byte) {
```

Comment:
The `chan<-` syntax for the type seems nice. We should change up this in the other places we have used channels as well.